rmf_fleet_adapter/src/rmf_fleet_adapter/tasks/Compose.cpp

```diff
@@ -126,10 +126,11 @@ void add_compose(
           errors.end(),
           phase_activity.errors.begin(), phase_activity.errors.end());
 
-        std::vector<rmf_task_sequence::Phase::ConstDescriptionPtr> cancel;
-        const auto& on_cancel_json = phase_json["on_cancel"];
-        if (on_cancel_json.is_array())
+        std::vector<rmf_task_sequence::Phase::ConstDescriptionPtr> cancel = {};
+        const auto cancel_it = phase_json.find("on_cancel");
+        if (cancel_it != phase_json.end() && cancel_it.value().is_array())
```

Comment:
We should not silently ignore `on_cancel` if it doesn't map to an array.

We can skip adding the cancellation sequence, but let's at least add a message to the `errors` vector so that the operators can be informed of the issue.